crawl-ref/source/fight.cc

```diff
@@ -751,7 +751,7 @@ void attack_cleave_targets(actor &attacker, list<actor*> &targets,
     while (attacker.alive() && !targets.empty())
     {
         actor* def = targets.front();
-        if (def && def->alive() && !_dont_harm(attacker, *def))
+        if (def && def->alive() && !_dont_harm(attacker, *def) && cleave_target_adjacent(attacker, *def))
```

Comment:
This doesn't really need a whole new function. Just use [`adjacent`](https://github.com/crawl/crawl/blob/dfd3b10bd21a7542d5e07d0e597264ac8ba37924/crawl-ref/source/coord.cc#L22).